### ISSUE TITLE
Exception propagation in BulkheadAsynchRetryTest

### ISSUE THREAD

**Issue by spericas** [eclipse/microprofile-fault-tolerance#357]
The spec states,

"A call to a method annotated with @Asynchronous will never throw an exception directly. 
Instead, the returned Future or CompletionStage will report that its task failed with the 
exception which would have been thrown.

For example, if @Asynchronous is used with @Bulkhead on a method which returns a Future 
and the bulkhead queue is full when the method is called, the method will return a Future 
where calling isDone() returns true and calling get() will throw an ExecutionException 
which wraps a BulkheadException."

 Some tests in `BulkheadAsynchRetryTest` rely on method `Utils::handleResults(int, Future[])` to monitor `Future`'s, yet this method does not propagate exceptions:

 ```
catch (Throwable e) {
    log(e.toString());		// logging only here ignores ExecutionException
 }
```	

 so in our implementation the expected `ExecutionException` wrapping the `BulkheadException` is ignored, causing the test to fail. Shouldn't these exceptions be propagated here?

Tests that seem to be affected by this behavior are:

```
   <include name="testBulkheadClassAsynchronous55RetryOverload"/>
   <include name="testBulkheadMethodAsynchronous55RetryOverload”/>
   <include name="testBulkheadMethodAsynchronousRetry55Trip”/>
```

**Comment by Emily-Jiang** [comment]
These tests expect the calling method to throw exception, which is incorrect assumption. I am ok to exclude these tests.

**Comment by spericas** [comment]
@Emily-Jiang I'm fine with excluding them, but propagating the exception reported by a `Future` as explained above should work as well.

**Comment by spericas** [comment]
Hi @hutchig, thanks for responding. Yes, we thought to start with 1.0 API and TCK and then move to the latest version. Sounds like we should just move up right away. 

Which version is the recommended one? 1.1.3?

**Comment by Emily-Jiang** [comment]
@spericas 1.1.3 is the latest released TCK version. I can exclude these 3 tests as we don't want to force implementations to directly throw exceptions from main thread. Also, as @Azquelt pointed out, these tests were rewritten for CompletionStage support in the current master branch.

**Comment by Emily-Jiang** [comment]
fixed and released 1.1.4
